server/src/instant/storage/s3.clj

```diff
@@ -90,11 +94,32 @@
 ;; Instant <> S3 integration
 ;; ----------------------
 
-(defn upload-file-to-s3 [{:keys [app-id location-id] :as ctx} file]
-  (when (not (instance? java.io.InputStream file))
+(def ^Tika tika (Tika.))
+
+(defn requires-tika? [{:keys [content-type app-id]}]
+  (and
+   (not content-type)
+   (contains? (flags/flag :tika-enabled-apps) app-id)))
+
+(defn deduce-content-type [{:keys [content-type] :as ctx} file]
+  (if (requires-tika? ctx)
+    (.detect tika ^TikaInputStream file)
+    content-type))
+
+(defn get-file-stream [ctx file]
+  (if (requires-tika? ctx)
+    (TikaInputStream/get ^InputStream file)
+    file))
+
+(defn upload-file-to-s3 [{:keys [app-id location-id] :as ctx} _file]
```

Comment:
If you put an underscore in front of `file`, then that tells the linter that it shouldn't report an error if you don't use the argument. I'd put it at the end (i.e. `file_`) so that the linter will complain if we forget to use it.